os-observer/src/events/npm.ts

```diff
@@ -1,37 +1,48 @@
 import dayjs, { Dayjs } from "dayjs";
 import _ from "lodash";
 import npmFetch from "npm-registry-fetch";
-import { EventSourceFunction, ApiReturnType } from "../utils/api.js";
-import { safeCast } from "../utils/common.js";
-import { logger } from "../utils/logger.js";
-import { parseGithubUrl } from "../utils/parsing.js";
 import {
-  prisma,
   ArtifactNamespace,
   ArtifactType,
   EventType,
+  getEventSourcePointer,
+  insertData,
+  prisma,
 } from "../db/prisma-client.js";
+import {
+  EventSourceFunction,
+  ApiInterface,
+  ApiReturnType,
+  CommonArgs,
+} from "../utils/api.js";
+import { assert, safeCast } from "../utils/common.js";
 import { InvalidInputError } from "../utils/error.js";
-import { assert } from "../utils/common.js";
+import { logger } from "../utils/logger.js";
+import { parseGithubUrl } from "../utils/parsing.js";
 
 // API endpoint to query
 const NPM_HOST = "https://api.npmjs.org/";
 // npm was initially released 2010-01-12
-const DEFAULT_START_DATE = "2010-01-01";
```

Comment:
For github I am setting the start date for the events pointer to be the time the repository was created. Is It possible to get something similar for NPM?